cli/pcluster/config/validators.py

```diff
@@ -1420,3 +1430,22 @@ def validate_backup_options(
         fsx_imported_file_chunk_size or fsx_import_path or fsx_export_path or fsx_auto_import_policy
     ) and fsx_automatic_backup_retention_days:
         errors.append("Backups cannot be created on S3-linked file systems")
+
+
+def validate_storage_type_options(
+    errors, fsx_storage_type, fsx_deployment_type, fsx_per_unit_storage_throughput, fsx_drive_cache_type
+):
+    if fsx_storage_type == "HDD":
+        if fsx_deployment_type != "PERSISTENT_1":
+            errors.append("For HDD filesystems, deployment_type must be 'PERSISTENT_1'")
+        if fsx_per_unit_storage_throughput not in [12, 40]:
+            errors.append("For HDD filesystems, per_unit_storage_throughput can only have values of 12 or 40")
+        if not fsx_drive_cache_type:
+            errors.append("For HDD filesystems, the 'drive_cache_type' must be specified")
+    if fsx_storage_type != "HDD" and fsx_drive_cache_type is not None:
+        errors.append("'drive_cache_type' features can be used only with HDD filesystems")
+    if not fsx_drive_cache_type and fsx_drive_cache_type is not None:
+        errors.append("'drive_cache_type' features can be used only with HDD filesystems")
```

Comment:
This condition is to avoid customer have an input for fsx_drive_cache_type = "NONE" without specifying the "Storage_type"

if not None: True
if None is not None: false